### PR TITLE
Add x-client-app and x-client-ver infrastructure [978]

### PR DESCRIPTION
**Pull request by MarkZuber**
https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/978

### CODE REVIEW

File: src/Microsoft.Identity.Client/AppConfig/AbstractApplicationBuilder.cs [hunk 1]
```diff
@@ -230,10 +253,10 @@ protected T WithOptions(ApplicationOptions applicationOptions)
         /// </summary>
         /// <param name="component">identifier of the software component (libraries/SDK) consuming MSAL.NET</param>
         /// <returns>The builder to chain the .With methods</returns>
+        [Obsolete("Should use ClientName and ClientVersion properties instead of Component", true)]
         public T WithComponent(string component)
```
Comment (jmprieur) [inline src/Microsoft.Identity.Client/AppConfig/AbstractApplicationBuilder.cs]: Should it be in Migration Aid ?
or otherwise debugger hidden?

Comment (bgavrilMS) [reply src/Microsoft.Identity.Client/AppConfig/AbstractApplicationBuilder.cs]: Or just remove them, i reckon there are very few developers using this (possibly 1-2 internal ones?)

File: src/Microsoft.Identity.Client/Microsoft.Identity.Client.csproj [hunk 2]
```diff
@@ -18,10 +18,12 @@
   <PropertyGroup Condition="'$(TargetFramework)' == '$(TargetFrameworkNetStandard)'">
     <DefineConstants>$(DefineConstants);NETSTANDARD_RUNTIME</DefineConstants>
   </PropertyGroup>
+  <ItemGroup>
```
Comment (bgavrilMS) [inline src/Microsoft.Identity.Client/Microsoft.Identity.Client.csproj]: Is this needed ? I don't see any resources in the Exceptions folder.

Comment (MarkZuber) [reply src/Microsoft.Identity.Client/Microsoft.Identity.Client.csproj]: no, this is VS being "helpful" again.  fixed

File: src/Microsoft.Identity.Client/AppConfig/AbstractApplicationBuilder.cs (hunk 1 above)
Comment (MarkZuber) [reply src/Microsoft.Identity.Client/AppConfig/AbstractApplicationBuilder.cs]: I've just deleted WithComponent then.

**Pull request merged**